multidim-interop/rust/src/lib.rs

```diff
@@ -1,14 +1,44 @@
+use std::time::Duration;
+
 use anyhow::{Context, Result};
 use env_logger::Env;
-use libp2pv0500::PeerId;
 use log::info;
 use redis::{AsyncCommands, Client as Rclient};
-use std::{env, time::Duration};
+use strum::EnumString;
 
 const REDIS_TIMEOUT: usize = 10;
 
+/// Supported transports by rust-libp2p.
+#[derive(Clone, Debug, EnumString)]
+#[strum(serialize_all = "kebab-case")]
+pub enum Transport {
+    Tcp,
+    QuicV1,
+    Webrtc,
+    Ws,
+}
+
+/// Supported stream multiplexers by rust-libp2p.
+#[derive(Clone, Debug, EnumString)]
+#[strum(serialize_all = "kebab-case")]
+pub enum Muxer {
+    Mplex,
+    Yamux,
+    Quic,
+    Webrtc,
```

Comment:
These aren't strictly muxers. I think we should leave the muxer variable empty when we don't explicitly set one. IMO, that will be cleaner implementation.

Reply:
yeah makes sense, thanks Thomas addressed :)